src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1534,2156 @@ interface MPSMatrixMultiplication {
+
+		[Export ("encodeToCommandBuffer:sourceMatrix:resultMatrix:pivotIndices:status:")]
+		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSMatrix sourceMatrix, MPSMatrix resultMatrix, MPSMatrix pivotIndices, [NullAllowed] IMTLBuffer status);
+
+		// inlining ctor from base class
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSMatrixUnaryKernel))]
+	[DisableDefaultCtor] // According to docs needs a Metal Device so initWithDevice: makes more sense.
+	interface MPSMatrixDecompositionCholesky {
+		[Export ("initWithDevice:lower:order:")]
+		IntPtr Constructor (IMTLDevice device, bool lower, nuint order);
+
+		[Export ("encodeToCommandBuffer:sourceMatrix:resultMatrix:status:")]
+		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSMatrix sourceMatrix, MPSMatrix resultMatrix, [NullAllowed] IMTLBuffer status);
+
+		// inlining ctor from base class
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MPSMatrixCopyDescriptor {
+		[Static]
+		[Export ("descriptorWithSourceMatrix:destinationMatrix:offsets:")]
+		MPSMatrixCopyDescriptor Create (MPSMatrix sourceMatrix, MPSMatrix destinationMatrix, MPSMatrixCopyOffsets offsets);
+
+		[Export ("initWithDevice:count:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, nuint count);
+
+		[Export ("setCopyOperationAtIndex:sourceMatrix:destinationMatrix:offsets:")]
+		void SetCopyOperation (nuint index, MPSMatrix sourceMatrix, MPSMatrix destinationMatrix, MPSMatrixCopyOffsets offsets);
+
+		[Export ("initWithSourceMatrices:destinationMatrices:offsetVector:offset:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSMatrix[] sourceMatrices, MPSMatrix[] destinationMatrices, [NullAllowed] MPSVector offsets, nuint byteOffset);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel))]
+	[DisableDefaultCtor] // There is a DesignatedInitializer, file a bug if needed.
+	interface MPSMatrixCopy {
+		[Export ("initWithDevice:copyRows:copyColumns:sourcesAreTransposed:destinationsAreTransposed:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, nuint copyRows, nuint copyColumns, bool sourcesAreTransposed, bool destinationsAreTransposed);
+
+		[Export ("copyRows")]
+		nuint CopyRows { get; }
+
+		[Export ("copyColumns")]
+		nuint CopyColumns { get; }
+
+		[Export ("sourcesAreTransposed")]
+		bool SourcesAreTransposed { get; }
```

Comment:
Mmh according to https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members we should probably use a prefix so `AreSourcesTransposed`.

> DO name Boolean properties with an affirmative phrase (CanSeek instead of CantSeek). Optionally, you can also prefix Boolean properties with "Is," "Can," or "Has," but only where it adds value.

If we do that we also need to update https://github.com/xamarin/xamarin-macios/pull/3005/files#diff-d05bb29f8320ca898f34fbaeb9d0385dR1842 and https://github.com/xamarin/xamarin-macios/pull/3005/files#diff-d05bb29f8320ca898f34fbaeb9d0385dR1854